src/MacVim/MMVimController.m

```diff
@@ -1306,7 +1355,53 @@ - (void)addToolbarItemWithLabel:(NSString *)label
 
     [toolbar insertItemWithItemIdentifier:label atIndex:idx];
 }
-
+#if MAC_OS_X_VERSION_MAX_ALLOWED > MAC_OS_X_VERSION_10_12
+- (void)addTouchbarItemWithLabel:(NSString *)label
+                           icon:(NSString *)icon
+                        atIndex:(int)idx
+{
+    // Check for separator items.
+    if (!label) {
+        label = NSTouchBarItemIdentifierFixedSpaceLarge;
+    } else if ([label length] >= 2 && [label hasPrefix:@"-"]
+                                   && [label hasSuffix:@"-"]) {
+        // The label begins and ends with '-'; decided which kind of separator
+        // item it is by looking at the prefix.
+        if ([label hasPrefix:@"-space"]) {
+            label = NSTouchBarItemIdentifierFixedSpaceSmall;
+        } else if ([label hasPrefix:@"-flexspace"]) {
+            label = NSTouchBarItemIdentifierFlexibleSpace;
+        } else {
+            label = NSTouchBarItemIdentifierFixedSpaceLarge;
+        }
+    } else {
```

Comment:
This whole `else` block has mixed tab/space indenting and seemingly overly deep indents.